pkg/ingester/ingester.go

```diff
@@ -1067,6 +1077,35 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 	return nil
 }
 
+func (i *Ingester) checkRateLimiting(userID string) (bool, error) {
+	if !i.cfg.DynamicRateLimitingEnabled {
+		return true, nil
+	}
+
+	// here we can directly call the ConcurrencyLimiter to check if we should reject request.
+	ok := i.concurrencyLimiter.Acquire(userID)
```

Comment:
need to return defer which would do onSuccess or OnDrop in order to release the token